src/components/CodeInput/styled/index.js

```diff
@@ -77,8 +77,12 @@ const StyledInput = attachThemeAttrs(styled.input)`
         :focus,
         :active {
             box-shadow: none;
-            background-color: ${props.palette.background.main};
-            border: 1px solid ${props.palette.border.main};
+
+            ${!props.error &&
```

Comment:
Since using `!` in conditions is more difficult to read, we should avoid using it on conditions.

Reply:
it makes the code more complex and difficult for developer's understanding

Reply:
sure